### PR TITLE
Fix protobuf vuln

### PR DESCRIPTION
**Pull request by akhalina**
The package github.com/gogo/protobuf@v1.1.1 specifically, but also apparently all of the gogo/protobuf versions have vulnerabilities. These protobuf packages are coming from the gobuffalo packages we carry in this repo. 

I removed our direct dependency on the github.com/gobuffalo/packr library by updating one of the tests to read in files manually. 

I know that gogen is used to generate mocks for some of our structs, but we don't explicitly depend on it. If someone needs to re-generate the mocks, you need to get the "go generate" command locally and use it re-generate them. We don't explicitly do it in any pipeline steps or anything so I think it's okay to remove them as a dependency especially if they are bringing in outdated and problematic packages.

Please let me know if I'm missing something here and we do in fact need some of these packages for some reason.

### CODE REVIEW

File: tests/new_test.go [hunk 1]
```diff
@@ -18,31 +17,35 @@ type tc struct {
...
-			Spec:    data.Bytes("passthroughenabled.yaml"),
+			File:    "passthroughenabled.yaml",
 			WantErr: false,
 		},
 	}
 
 	for _, tt := range tcs {
 		t.Run(tt.Name, func(t *testing.T) {
-			_, err := transportd.New(
+			fileInput, err := ioutil.ReadFile(folder + tt.File)
```
Comment (kconwayinvision) [inline tests/new_test.go:42]: nit: You won't hit this with the way the test is set up but I'd recommend using `path.Join` to handle edge cases related to too many or too few path separators in the strings. For example, `./folder + missingruntime.yaml` would result in a test failure because of a missing slash.

Also, if you prefer to keep the content embedded in the binary instead of needing access to disk for the test then you can use the new `go:embed` directive: https://golang.org/pkg/embed/. It can expose either a string or a filesystem interface for loading files. You'd add something like
```golang
//go:embed ./specs/*
var specs embed.FS
```
and would be able to reference files like:
```golang
b, err := specs.ReadFile(tt.File)
```

**Comment by akhalina** [comment]
Realized I only ran unit tests locally, fixing integration tests now :)

**Review by alecbert-atl — APPROVED** [review]
Love that go.sum diff!

**Review by gcase555 — DISMISSED** [review]

File: tests/new_test.go [hunk 2]
```diff
@@ -4,45 +4,54 @@ package inttest
...
 
 type tc struct {
 	Name    string
-	Spec    []byte
+	File    string
 	WantErr bool
 }
 
+//go:embed specs
+var specs embed.FS
```
Comment (gcase555) [inline tests/new_test.go:22]: :interesting:

File: tests/new_test.go (hunk 1 above)
Comment (akhalina) [reply tests/new_test.go]: Good callout on the path join, I somehow always forget to use it. 

The embed is cool! But it took me longer than I want to admit to realize that it still wants a full relative path from this file, instead of a path relative to the folder you tell it to embed.

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/asecurityteam/transportd
 
-go 1.13
+go 1.16
```
Comment (gcase555) [inline go.mod:3]: this is a worthy upgrade but consumers of transportd should be aware they'll need to bump their go versions as well. We can possibly signify this via a major version upgrade of the version number(git tag).

**Review by gcase555 — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (akhalina) [reply go.mod:3]: I can do a major version upgrade. If it makes you feel any better sdcli is already on v1.16 of golang so that shouldn't be an issue (which is why the integrations tests run fine here). This is a minor golang version upgrade so I don't think any other upgrades should cause any issues.

Comment (gcase555) [reply go.mod:3]: Yeah in this case I think a major version upgrade is warranted. We do still have some older apps, like kvstore-cidr-aws, that are go 1.12 and import transportd in go.mod. 

Ideally sdcli would match an apps version but its technically used via a separate container and not imported into the code directly so might not be as big a deal, some of these older apps might still be pointing to an older sdcli version.

Comment (kconwayinvision) [reply go.mod:3]: Because Go is backwards compatible and upgrading is considered safe in virtually all cases, most projects don't do a major rev when updating the version of Go they use. For example, the older service should be able to upgrade to 1.16 at the same time as upgrading the library without any negative effects. On the other hand, the Go module system sort of discourages against non-v1 modules. For example, going v2 without embedding all the content in a v2 subdirectory breaks the typical relationship between the import path and the source code layout. It also requires changing both the module path to include a `/v2` suffix in addition to using a `v2.x.x` tag naming scheme. None of it is really a big deal as much as bit of a chore.

That being said, I'm not particularly picky about it and can easily adjust my `go.mod` to bring in a v2.

Comment (gcase555) [reply go.mod:3]: If migrating to 1.16 is not a major change then I am totally fine with a minor version bump here.

**Review by ensary — APPROVED** [review]
